### PR TITLE
[ts-sdk] Add `useInputStreams` hook

### PR DESCRIPTION
**Pull request by wkozyra95**
- Add `useInputStreams` to interact with new streams
- Add lerna + improve build/release automation (e.g. running npm run watch for all workspaces)
- Ensure package is build when publishing

### CODE REVIEW

File: ts/@live-compositor/core/package.json [hunk 1]
```diff
@@ -6,12 +6,16 @@
   "scripts": {
     "lint": "eslint .",
     "typecheck": "tsc --noEmit",
-    "watch": "tsc --watch",
+    "watch": "tsc --watch --preserveWatchOutput",
```
Comment (wkozyra95) [inline ts/@live-compositor/core/package.json:9]: Terminal will not be cleared after file is saved and errors/warnings are printed again. This is needed to stream `npm run output` from multiple commands

File: ts/@live-compositor/core/package.json [hunk 2]
```diff
@@ -6,12 +6,16 @@
...
     "build": "tsc",
-    "clean": "rimraf dist"
+    "clean": "rimraf dist",
+    "prepublishOnly": "npm run clean && npm run build"
   },
   "author": "",
   "license": "MIT",
+  "files": [
+    "/dist"
```
Comment (wkozyra95) [inline ts/@live-compositor/core/package.json:17]: only include dist directory to minimize size of the package

File: ts/@live-compositor/core/src/compositor.ts [hunk 3]
```diff
@@ -14,6 +14,7 @@ export async function createLiveCompositor(manager: CompositorManager): Promise<
 export class LiveCompositor {
   private manager: CompositorManager;
   private api: ApiClient;
+  private store: ContextStore = new ContextStore();
```
Comment (wkozyra95) [inline ts/@live-compositor/core/src/compositor.ts:17]: this value is shared between all outputs

**Review by noituri — APPROVED** [review]

File: ts/@live-compositor/core/src/compositor.ts (hunk 3 above)
Comment (WojciechBarczynski) [inline ts/@live-compositor/core/src/compositor.ts:17]: [NIT]
Personal opinion here, but `store` as field name sounds weird. It doesn't indicate what is "stored" there. I would use `context`.

File: ts/@live-compositor/core/src/output.ts [hunk 4]
```diff
@@ -1,57 +1,91 @@
...
-      this.audioOptions = intoAudioInputsConfiguration(output.audio.initial);
+      audioOptions = intoAudioInputsConfiguration(output.audio.initial);
     }
-    this.initialized = true;
+    this.throttledUpdate = throttle(async () => {
+      await api.updateScene(this.outputId, {
+        video: this.videoRenderer && { root: this.videoRenderer.scene() },
+        audio: audioOptions,
+      });
+    }, 30);
```
Comment (WojciechBarczynski) [inline ts/@live-compositor/core/src/output.ts:35]: Where `30` come from? With 60FPS per frame updates happen every 16ms.

**Review by WojciechBarczynski — APPROVED** [review]

File: ts/@live-compositor/core/src/output.ts (hunk 4 above)
Comment (wkozyra95) [reply ts/@live-compositor/core/src/output.ts:35]: > Where 30 come from? 

I think this is optimal value for limiting the updates

> With 60FPS per frame updates happen every 16ms.

I agree with the math, but I don't see what point you are making.

File: ts/@live-compositor/core/src/compositor.ts (hunk 3 above)
Comment (wkozyra95) [reply ts/@live-compositor/core/src/compositor.ts:17]: this is store, context it's what returned from the hooks

File: ts/@live-compositor/core/src/output.ts (hunk 4 above)
Comment (WojciechBarczynski) [reply ts/@live-compositor/core/src/output.ts:35]: With this throttling, it's impossible to do per-frame updates in 60FPS.

File: ts/@live-compositor/core/src/compositor.ts (hunk 3 above)
Comment (WojciechBarczynski) [reply ts/@live-compositor/core/src/compositor.ts:17]: What do you mean by `store`? Every data structure can be referred to as a `store`. Every struct, map, list, etc. "stores" sth. My point is that this name doesn't indicate what is stored there. 

When you see:
```
compositor.store
```
It's impossible to tell what is in `store` without checking this `LiveCompositor` class definition/field type.

File: ts/@live-compositor/core/src/output.ts (hunk 4 above)
Comment (wkozyra95) [reply ts/@live-compositor/core/src/output.ts:35]: I don't think it should be possible to do any per frame stuff, there is no way to synchronize it either way

File: ts/@live-compositor/core/src/compositor.ts (hunk 3 above)
Comment (wkozyra95) [reply ts/@live-compositor/core/src/compositor.ts:17]: - this internal implementation, this is not user facing
- store is well established name in react and probably even in general in web ecosystem

**Pull request merged**